modules/table/src/test/java/org/apache/ignite/internal/table/distributed/replicator/ReplicatorUtilsTest.java

```diff
@@ -0,0 +1,127 @@
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.UUID;
+import java.util.function.Consumer;
+import org.apache.ignite.internal.catalog.CatalogManager;
+import org.apache.ignite.internal.catalog.CatalogService;
+import org.apache.ignite.internal.catalog.CatalogTestUtils;
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.hlc.HybridClockImpl;
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.apache.ignite.internal.table.distributed.replication.request.ReadWriteReplicaRequest;
+import org.apache.ignite.internal.testframework.IgniteAbstractTest;
+import org.junit.jupiter.api.Test;
+
+/** For {@link ReplicatorUtils} testing. */
+public class ReplicatorUtilsTest extends IgniteAbstractTest {
+    private final HybridClock clock = new HybridClockImpl();
+
+    @Test
+    void testBeginRwTxTs() {
+        HybridTimestamp beginTs = clock.now();
+
+        UUID txId = transactionId(beginTs, 10);
+
+        assertEquals(beginTs, beginRwTxTs(readWriteReplicaRequest(txId)));
+    }
+
+    @Test
+    void testRwTxActiveCatalogVersion() {
+        HybridTimestamp beginTs = clock.now();
+
+        UUID txId = transactionId(beginTs, 10);
+
+        CatalogService catalogService = mock(CatalogService.class);
+
+        when(catalogService.activeCatalogVersion(anyLong())).thenReturn(666);
+
+        assertEquals(666, rwTxActiveCatalogVersion(catalogService, readWriteReplicaRequest(txId)));
+
+        verify(catalogService).activeCatalogVersion(eq(beginTs.longValue()));
+    }
+
+    @Test
+    void testLatestIndexDescriptorInBuildingStatus() throws Exception {
+        withCatalogManager(catalogManager -> {
+            createSimpleTable(catalogManager, TABLE_NAME);
+
+            int tableId = getTableIdStrict(catalogManager, TABLE_NAME, clock.nowLong());
+
+            assertNull(latestIndexDescriptorInBuildingStatus(catalogManager, tableId));
+
+            createSimpleHashIndex(catalogManager, TABLE_NAME, INDEX_NAME);
+            assertNull(latestIndexDescriptorInBuildingStatus(catalogManager, tableId));
+
+            int indexId = getIndexIdStrict(catalogManager, INDEX_NAME, clock.nowLong());
+
+            startBuildingIndex(catalogManager, indexId);
+            assertEquals(indexId, latestIndexDescriptorInBuildingStatus(catalogManager, tableId).id());
+
+            makeIndexAvailable(catalogManager, indexId);
+            assertEquals(indexId, latestIndexDescriptorInBuildingStatus(catalogManager, tableId).id());
+
+            createSimpleHashIndex(catalogManager, TABLE_NAME, INDEX_NAME + "_ONE_MORE");
+            assertEquals(indexId, latestIndexDescriptorInBuildingStatus(catalogManager, tableId).id());
```

Comment:
How about also starting the build of the second index and making sure that `latestIndexDescriptorInBuildingStatus()` will return the second index this time?

Reply:
fix it